quadriga/src/main/java/edu/asu/diging/quadriga/api/v1/AddNetworkApiController.java

```diff
@@ -44,7 +50,13 @@
      */
     @ResponseBody
     @RequestMapping(value = "/api/v1/network/add", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE)
-    public HttpStatus processJson(@RequestBody Quadruple quadruple) {
+    public HttpStatus processJson(@RequestBody Quadruple quadruple, @RequestParam(value = "collectionid", required = true) String collectionId) {
+        
+        Collection collection = collectionManager.findCollection(collectionId);
+        
+        if(collection == null) {
+            return HttpStatus.NOT_ACCEPTABLE;
```

Comment:
This status code is in cases of: "The requested resource is capable of generating only content not acceptable according to the Accept headers sent in the request."

Here it should either return a not found (if the collection is part of the url) or bad request (if the request is a request parameter)